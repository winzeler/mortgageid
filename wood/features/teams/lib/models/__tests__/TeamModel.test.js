const { TeamModel } = require('#features/teams/lib/models/TeamModel');

describe('TeamModel', () => {
  describe('hasFlag', () => {
    it('should test to see if team has flags', () => {
      const team = new TeamModel({ flags: {
        aFlag: 'somevalue',
        specificFlag: 'specific_value',
      } });

      expect(team.hasFlag('aFlag')).toBe(true);
      expect(team.hasFlag('specificFlag', 'specific_value')).toBe(true);
      expect(team.hasFlag('specificFlag', 'other_value')).toBe(false);
    });

    // ---------------------------------------------------------------------------------------------

    it('should test for a secure flags', () => {
      const team = new TeamModel({ secure_flags: {
        aFlag: 'somevalue',
        specificFlag: 'specific_value',
      } });

      expect(team.hasFlag('aFlag')).toBe(true);
      expect(team.hasFlag('specificFlag', 'specific_value')).toBe(true);
      expect(team.hasFlag('specificFlag', 'other_value')).toBe(false);
    });

    // ---------------------------------------------------------------------------------------------

    it('should not serialize secure flags', () => {
      const team = new TeamModel({
        flags: { insecureFlag: 'insecure_value' },
        secure_flags: { secureFlag: 'secure_value' },
      });

      expect(team.toJSON()).toMatchSnapshot({
        created_at: expect.any(String),
        updated_at: expect.any(String),
      });
    });
  });
});
