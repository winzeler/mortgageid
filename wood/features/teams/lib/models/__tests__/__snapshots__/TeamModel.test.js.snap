// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamModel hasFlag should not serialize secure flags 1`] = `
Object {
  "created_at": Any<String>,
  "currency": undefined,
  "flags": Object {
    "insecureFlag": "insecure_value",
  },
  "id": undefined,
  "name": undefined,
  "subscription": undefined,
  "updated_at": Any<String>,
}
`;
